compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2460,6 +2471,12 @@ private boolean hasSameReadonlyFlag(BType source, BType target) {
         }
 
         public Boolean visit(BTupleType t, BType s) {
+            if (((!t.tupleTypes.isEmpty() && checkAllTupleTypeMembersBelongNoType(t)) ||
```

Comment:
We can probably directly pass `t.tupleTypes` to `checkAllTupleTypeMembersBelongNoType`, right?

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/issues/28839